botocore/endpoint.py

```diff
@@ -302,10 +302,19 @@ def _do_get_response(self, request, operation_model, context):
         history_recorder.record('HTTP_RESPONSE', http_response_record_dict)
 
         protocol = operation_model.metadata['protocol']
+        additional_response_keys = {}
+        self._event_emitter.emit(
+            f"before-parse.{service_id}.{operation_model.name}",
+            **{
+                'response_dict': response_dict,
+                'additional_response_keys': additional_response_keys,
+            },
+        )
         parser = self._response_parser_factory.create_parser(protocol)
         parsed_response = parser.parse(
             response_dict, operation_model.output_shape
         )
+        parsed_response.update(additional_response_keys)
```

Comment:
yes, `response_dict` doesn't persist because we use this dict as a temporary variable to help us populate `parsed_response` with all the modeled attributes. after that, we discard the `response_dict` variable.